### DISCUSSION TITLE
I made a Spotify Store version with SpotX integrated (outdated)

### DISCUSSION THREAD

**Discussion by raziel6868** [SpotX-Official/SpotX#113] [General]
### Description of the new feature / enhancement

I heard that UWP (Microsoft Store apps) benefits from CPU/GPU optimization work done by Microsoft, so I integrated SpotX (full installation mode) into the Store version of Spotify.
How to use: Unzip file, run Install.cmd as administrator and enjoy 😊.
Link: https://drive.google.com/file/d/1Cn8xdJe9ok04nZ9QBEJCI5mvSaWcVT7f/view?usp=sharing
**P/S: Please uninstall the official Spotify Microsoft Store version before installing this modified version**

**Comment by amd64fox** [comment]
If I understand correctly, then Spotify from MS and the regular desktop version are no different except for their packaging, so why should the 1st work faster?

**Reply by raziel6868** [reply]
Your guess is right, the UWP/Microsoft Store version is basically a regular Win32/Desktop version secured by three files: AppxBlockMap.xml, AppxManifest.xml, and AppxSignature.p7x. If there is a modification in the MSIX/Appx package that does not match the information in the three files, or if one of the three files is deleted, the app will fail to run in UWP mode; however, you can still run it in Win32 mode by double clicking on the program at C:\Program Files\WindowsApps. You must create a new MSIX/Appx package containing the changes you made, and then install the MSIX/Appx package to run the app in UWP mode.

I'm not sure if the Store version is faster than the desktop version, but as a college student concerned about quickly running out of battery on my laptop, I'd go with the UWP app. This was mentioned in these Reddit posts:
- https://www.reddit.com/r/AcerNitro/comments/rfwjah/how_i_achieved_12_hours_battery_lifeguide/
- https://www.reddit.com/r/thinkpad/comments/dddsqv/guide_ultimate_windows_battery_life_tips/

**Reply by amd64fox** [reply]
You misunderstood, Spotfiy MS app and Spotfiy Desktop are the same app, it was just repackaged. Therefore, I think you will not be able to save the battery of your laptop on this, although I may be wrong.

Try to compare 3 versions of Spotfiy:
- Web
- MS
- Desktop

Just measure in the same conditions who drains the battery faster, for example, in an hour or in 3 hours.

**Reply by raziel6868** [reply]
When you double-click the software in the root folder, C:\Program Files\WindowsApps, it will run in Win32 mode, allowing the software to fully utilize the machine's performance. However, if you click on the "perfectly" square icon generated by the operating system, as seen below, it will run in UWP mode, which means the software will function as a "phone app" because it has been sandboxed by Windows, resulting in battery savings.

![Screenshot 2022-08-16 224651](https://user-images.githubusercontent.com/87901700/184922795-4354f3dc-e2c8-44e0-beb8-cfde70ab801a.png)

![Screenshot 2022-08-16 224356](https://user-images.githubusercontent.com/87901700/184922315-bbadbf9e-c210-40c8-8a2b-b34d88c601d0.png)

**Reply by amd64fox** [reply]
Think for yourself, we take 2 identical applications, we run one in the win 10 sandbox, and the other we just run in win 10, then according to your assumption, the first one will somehow spend less resources than the second one.
Do you really think so ? )

I would check it in practice, by means of the test above, in theory everything is always good.

**Reply by amd64fox** [reply]
Here they write in the comments about what I'm trying to tell you
https://www.reddit.com/r/Windows10/comments/lkxp7o/when_i_have_the_choice_should_i_install_an_app/

**Comment by simysighs** [comment]
do you still have the files on you? I would like them if you still do.

**Reply by raziel6868** [reply]
> do you still have the files on you? I would like them if you still do.

dm me on discord KawaiiBalls#6732

**Reply by lennylulw** [reply]
do  any of you have the files please?

**Comment by HotarunIchijou** [comment]
How can I get this?

**Comment by EdwardDK** [comment]
Can you please update the link. I want to se this for xbox :)

**Reply by EdwardDK** [reply]
DM me on discord: .eddy47

**Comment by tayfrm99** [comment]
Can you update the link? I would like to see it in action and possibly test the performance

edit: dm me the file on discord: mushiikaaa

**Comment by lennylulw** [comment]
@raziel6868 hey do you still have the files for it? dm me on dc if so thanks. lenny_haha
